packages/wonder-blocks-core/src/components/__tests__/with-ssr-placeholder.test.js

```diff
@@ -186,4 +187,73 @@ describe("WithSSRPlaceholder", () => {
             });
         });
     });
+
+    describe("inside a RenderStateRoot", () => {
+        test("calls placeholder render first, then the actual content render", async () => {
+            // Arrange
+            const mockPlaceholder = jest.fn(() => null);
+            await new Promise((resolve) => {
+                const nodes = (
+                    <RenderStateRoot>
+                        <WithSSRPlaceholder placeholder={mockPlaceholder}>
+                            {() => {
+                                resolve();
+                                return null;
+                            }}
+                        </WithSSRPlaceholder>
+                    </RenderStateRoot>
+                );
+
+                // Act
+                mount(nodes);
+            });
+
+            // Assert
+            // Our promise doesn't resolve until the children render, therefore
+            // we don't get here until that and so if the placeholder has been
+            // called, it must have been called first.
+            expect(mockPlaceholder).toHaveBeenCalledTimes(1);
```

Comment:
hmm I wonder if it would be clearer if we did something like
`const trace = []`;
`placeholder={() => trace.push("placeholder")}`
`{() => trace.push("content")}`

then we could assert the order more explicitly, with `expect(trace).toEqual(["placeholder", "content"])`

Reply:
I copied some of the existing tests so I'd like to stay with this style.